lms/envs/common.py

```diff
@@ -480,6 +480,7 @@
 OAUTH2_DEFAULT_SCOPES = {
     'read': _('Read access'),
     'write': _('Write access'),
+    'user_id': _('Know your user_id'),
```

Comment:
Reviewer: FYI: I couldn't get user_id scope to work with ecommerce without this change.

Reply:
The problem was that the SSO path is specifying the scope user_id, but the lms would error with invalid_scope. Maybe the ecommerce-sso application also needs the scope defined in the admin, but I wasn’t clear how that was done and couldn’t get it to work. Also, if we can get it to work by doing an application related config in admin, we would need to do this for all applications using auth-backends.

Or, maybe there is some other way to stop the scope validation from failing that I wasn’t able to find yet?  

This change at least let me test out the rest of the work. 🙂

Reply:
I believe there are 2 settings: 1 for DEFAULT scopes and 1 for ALLOWED scopes.  I don't remember the name of the latter.

Reply:
Thanks. I’ll look into this tomorrow.

Reply:
@nasthagiri: Now that I found what I was looking for, I can speak much more clearly about this now.  Above, when I wrote "an application related config in admin", I meant we would need to add the `user_id` scope in admin under [oauth_dispatch/applicationaccess](http://localhost:18000/admin/oauth_dispatch/applicationaccess/).  The latest code in this PR moves the scope from the defaults to the available scopes, and I had to add an application access record for ecommerce-sso that added back `user_id` to get it to work.

The following are the applications that get created in devstack:
![Screen Shot 2019-03-27 at 10 10 37 AM](https://user-images.githubusercontent.com/14576445/55084182-2d235300-507b-11e9-8af3-738fb23c3b64.png)

If we want to keep `user_id` from being a default scope, I see the following additional work:

1. The PR for edx/auth-backends would need to be adjusted to not always request `user_id`, but instead to request it via a new setting like [`EXTRA_SCOPES`](https://github.com/search?q=org%3Aedx+%22EXTRA_SCOPE%22&type=Code) which is used for EdXOpenIdConnect.  Note: since the scopes don't match, the new setting might be named something like `OAUTH2_EXTRA_SCOPES`.

2. We need to update many of the above pictured applications in all environments (devstack, sandboxes?, prod, etc.) to have the `user_id` scope added to oAuth_Dispatch / Application Access. I don't think we yet have a mechanism for this that would make this simpler, so that would need to be designed/implemented.

@nasthagiri: Do you agree with this assessment? Do you agree that I should move forward regarding all of this?

Thank you!

Reply:
For #1, I started by making `user_id` a requirement to the new oAuth+SSO flow in https://github.com/edx/auth-backends/pull/59 as you implied.

For #2, I'll look into it further.  Looks like `registrar` is also using [EdXOAuth2](https://github.com/search?q=org%3Aedx+EdXOAuth2&type=Code).

Reply:
I am going to resolve this conversation.  Any further action would be on https://github.com/edx/auth-backends/pull/59, and would not affect this PR.